src/main/java/org/mifos/connector/ams/zeebe/ZeebeeWorkers.java

```diff
@@ -589,6 +589,29 @@ public void setupWorkers() {
                         .name(WORKER_ACCOUNT_IDENTIFIER + dfspid)
                         .maxJobsActive(workerMaxJobs)
                         .open();
+
+                logger.info("## generating " + WORKER_SEND_CALLBACK + " worker");
+                zeebeClient.newWorker()
+                        .jobType(WORKER_SEND_CALLBACK)
+                        .handler((client, job) -> {
+                            logWorkerDetails(job);
+                            Map<String, Object> existingVariables = job.getVariablesAsMap();
+                            logger.debug("Exisiting variables {}", existingVariables);
+                                Exchange ex = new DefaultExchange(camelContext);
+                                Map<String, Object> variables = job.getVariablesAsMap();
+
+                                ex.setProperty(X_CALLBACKURL, existingVariables.get(X_CALLBACKURL));
+                                ex.setProperty(FINERACT_RESPONSE_BODY, existingVariables.get(FINERACT_RESPONSE_BODY));
+                                producerTemplate.send("direct:send-callback", ex);
+                                variables.put(CALLBACK_SUCCESS, ex.getProperty("callbackSent").toString());
```

Comment:
this may not be feasible with asynch mode of call-back calls. call back calls should not impact the performance of the app, if the calling server is slow.

Reply:
This will be handled by camel. If the exchange variable didn’t get callbackSent, it will send a callback again at least 3 times.